polygon/p2p/peer_shuffle.go

```diff
@@ -0,0 +1,15 @@
+package p2p
+
+import "math/rand"
+
+type PeerShuffle func(peerIds []*PeerId)
+
+func RandPeerShuffle(peerIds []*PeerId) {
+	rand.Shuffle(len(peerIds), func(i, j int) {
+		peerIds[i], peerIds[j] = peerIds[j], peerIds[i]
+	})
+}
```

Comment:
this looks very generic. could you please generalize it to work on any slice `[]T` and move to erigon-lib/common/collections.go ?

Reply:
ive added a generic func in erigon-lib/common/collections.go, however I would like to keep this interface and funcs here to aid with tests, now simply RandPeerShuffle calls common.SliceShuffle